packages/analytics/analytics-metric-provider/src/components/MetricsProviderInternal.vue

```diff
@@ -61,6 +61,18 @@ if (!queryBridge) {
   queryFn = queryBridge.queryFn
 }
 
+const analyticsConfigStore = useAnalyticsConfigStore()
+const analyticsConfig = analyticsConfigStore.getConfig()
```

Comment:
I would think at a minimum you would need to pass the Pinia instance from the host app into the `useAnalyticsConfigStore()` function in order to ensure it's using the same instance

Reply:
It works because the Pinia dependency is externalized, which means that it uses the host app's instance.  I should probably add something to the `config-store` README to explain that -- I'm basically following the discussion here: https://github.com/vuejs/pinia/discussions/1073#discussioncomment-2196098.  The pattern I'm aiming for is this:

- The host app calls `createPinia`.
- Later on, some component renders an analytics component.
- The analytics component needs to know the org's analytics config, so it calls `useAnalyticsConfigStore`.
- If the store hasn't already been created, it gets created.  Because the store is accessed during component render, it is guaranteed to happen after the host app has called `createPinia`.
- The analytics component calls `getConfig`, which hits the API to retrieve config.
- Later on, even if the component has been unmounted, the store remembers the config and makes it available to other components.

In other words, ideally the store is never even exposed to the host app.  At some point soon, I'm also planning to create an Analytics version of `AuthValidate` that would talk to this store to support various config checks without having to make the host app aware of the store.

I _really_ want to avoid passing the host app's Pinia instance down to the component explicitly because that would add additional boilerplate _somewhere_.  Within Pinia itself, global instance is managed via `provide` / `inject`, which feels like the right fit for what this is trying to accomplish.  However, I'd be happy to discuss further.

> I'm worried about the component instance having access to the same Pinia instance as the host app

I'm curious about this concern -- it feels like it shouldn't be an issue unless the host app wants to create a store with the exact same name, which seems unlikely?

Reply:
Updated the README.